packages/repack/src/webpack/plugins/ChunksToHermesBytecodePlugin/ChunksToHermesBytecodePlugin.ts

```diff
@@ -0,0 +1,224 @@
+      );
+      logger.info(
+        `Did you forget to provide an explicit '--bundle-output <path>' option to 'webpack-bundle'?`
+      );
+
+      throw new Error(
+        'Hermes compilation is only supported for bundle builds with --bundle-output'
+      );
+    }
+
+    const shouldUseHermesByteCode = (filename: string) =>
+      ModuleFilenameHelpers.matchObject(this.config, filename);
+
+    const hermesCLIPath =
+      this.config.hermesCLIPath || getHermesCLIPath(reactNativePath);
+
+    /** Directory where the bundle is saved */
+    const bundleOutputDir = path.dirname(bundleFilename);
+
+    /** Directory where assets are saved */
+    const assetsOutputDir = assetsPath || bundleOutputDir;
+
+    /** Directory where source maps are saved */
+    const sourcemapOutputDir = sourceMapFilename
+      ? path.dirname(sourceMapFilename)
+      : bundleOutputDir;
+
+    compiler.hooks.afterEmit.tapPromise(this.Name, async (compilation) => {
+      await Promise.all(
+        compilation
+          .getAssets()
+          .filter((asset) => shouldUseHermesByteCode(asset.name))
+          .map(async (asset) => {
+            /**
+             * Logic based on implementations for each platform.
+             * #### iOS
+             * - Logic in [react-native-xcode.sh](https://github.com/facebook/react-native/blob/f38fc9ba8681622f7cfdb586753e50c596946929/packages/react-native/scripts/react-native-xcode.sh#L166-L187)
+             *
+             * #### Android
+             * - Defaults in [ReactExtension.kt](https://github.com/facebook/react-native/blob/f38fc9ba8681622f7cfdb586753e50c596946929/packages/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/ReactExtension.kt#L116-L117)
+             * - Logic in [BundleHermesCTask.kt](https://github.com/facebook/react-native/blob/f38fc9ba8681622f7cfdb586753e50c596946929/packages/react-native-gradle-plugin/src/main/kotlin/com/facebook/react/tasks/BundleHermesCTask.kt#L93-L111)
+             */
+
+            logger.debug(
+              `Starting hermes compilation for asset: ${asset.name}`
+            );
+            const assetPath = compilation.getPath(asset.name);
+
+            // TODO: extract the following logic calculating paths to a separate
+            // function, share with OutputPlugin, or maybe integrate?
+            const isMainBundle = asset.name === 'index.bundle';
+
+            const bundleOutputPath = isMainBundle
+              ? bundleFilename
+              : path.join(
+                  platform === 'ios' ? assetsOutputDir : bundleOutputDir,
+                  assetPath
+                );
+
+            const bundleExists = await fileExists(bundleOutputPath);
+            if (!bundleExists) {
+              logger.error(`Bundle does not exist: ${bundleOutputPath}`);
+              // TODO: for now ignore this, consider throwing an error when remote
```

Comment:
We should solve running these transformations for remote assets. This is partially why I'd like to combine this logic with the asset processing that happens in the `OutputPlugin`.

For now this code just logs and skips the remote asset